src/commonMain/kotlin/operations/string/Substr.kt

```diff
@@ -0,0 +1,42 @@
+package operations.string
+
+import LogicOperation
+import asList
+import intOrZero
+
+object Substr : LogicOperation, StringUnwrapStrategy {
+    override val key: String = "substr"
+
+    override fun invoke(expression: Any?, data: Any?): String {
+        return with(expression.asList) {
+            val startIndex = getOrNull(1).toString().intOrZero
+            val endIndex = getOrNull(2).toString().intOrZero
+            substringOrEmpty(startIndex, endIndex)
+        }
+    }
+
+    private fun List<Any?>.substringOrEmpty(startIndex: Int, endIndex: Int): String {
+        val baseString = unwrapValues(firstOrNull()).joinToString(",")
+        return runCatching {
+            when {
+                size == 2 -> baseString.fromStartIndexToEnd(startIndex)
+                size >= 3 -> baseString.fromStartIndexToEndIndex(startIndex, endIndex)
+                else -> baseString
+            }
+        }.getOrNull().orEmpty()
+    }
+
+    private fun String.fromStartIndexToEnd(startIndex: Int) = if (startIndex > 0) {
+        substring(startIndex)
+    } else {
+        substring(length + startIndex)
+    }
+
+    private fun String.fromStartIndexToEndIndex(startIndex: Int, endIndex: Int) = when {
+        startIndex >= 0 && endIndex > 0 -> substring(startIndex, startIndex + endIndex)
```

Comment:
soo isn't `endIndex` in fact the length of substring we want to get, not endIndex?

Reply:
ye, youre right. I renamed it in postprocess. It should be named more like charsCount